apps/examples/testcase/ta_tc/task_manager/utc/utc_task_manager_main.c

```diff
@@ -406,25 +454,73 @@ static void utc_task_manager_broadcast_p(void)
 		TC_ASSERT_LEQ_CLEANUP("task_manager_broadcast", sleep_cnt, 10, sem_destroy(&tm_broad_sem));
 		sleep_cnt++;
 	}
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_on_cnt, TM_BROAD_TASK_NUM, sem_destroy(&tm_broad_sem));
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_off_cnt, 0, sem_destroy(&tm_broad_sem));
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_undefined_cnt, 0, sem_destroy(&tm_broad_sem));
 
 	broad_wifi_on_cnt = 0;
 	broad_wifi_off_cnt = 0;
+	broad_undefined_cnt = 0;
 	sleep_cnt = 0;
 	(void)task_manager_broadcast(TM_BROADCAST_WIFI_OFF);
+	while (1) {
+		usleep(500);
+		if (broad_wifi_off_cnt == TM_BROAD_TASK_NUM) {
+			break;
+		}
+		TC_ASSERT_LEQ_CLEANUP("task_manager_broadcast", sleep_cnt, 10, sem_destroy(&tm_broad_sem));
+		sleep_cnt++;
+	}
 	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_on_cnt, 0, sem_destroy(&tm_broad_sem));
 	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_off_cnt, TM_BROAD_TASK_NUM, sem_destroy(&tm_broad_sem));
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_undefined_cnt, 0, sem_destroy(&tm_broad_sem));
+
+	broad_wifi_on_cnt = 0;
+	broad_wifi_off_cnt = 0;
+	broad_undefined_cnt = 0;
+	sleep_cnt = 0;
+	(void)task_manager_broadcast(tm_broadcast_undefined_msg);
 	while (1) {
-		usleep(1);
-		if (broad_wifi_off_cnt == TM_BROAD_TASK_NUM) {
+		usleep(500);
+		if (broad_undefined_cnt == TM_BROAD_TASK_NUM) {
 			break;
 		}
 		TC_ASSERT_LEQ_CLEANUP("task_manager_broadcast", sleep_cnt, 10, sem_destroy(&tm_broad_sem));
 		sleep_cnt++;
 	}
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_on_cnt, 0, sem_destroy(&tm_broad_sem));
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_wifi_off_cnt, 0, sem_destroy(&tm_broad_sem));
+	TC_ASSERT_EQ_CLEANUP("task_manager_broadcast", broad_undefined_cnt, TM_BROAD_TASK_NUM, sem_destroy(&tm_broad_sem));
+
 	sem_destroy(&tm_broad_sem);
 	TC_SUCCESS_RESULT();
 }
 
+static void utc_task_manager_unset_broadcast_cb_n(void)
+{
+	int ret;
+
+	ret = task_manager_unset_broadcast_cb(TM_INVALID_BROAD_MSG, TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_unset_broadcast_cb", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_unset_broadcast_cb(TM_BROADCAST_WIFI_ON, TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_unset_broadcast_cb", ret, TM_INVALID_PARAM);
+
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_task_manager_unset_broadcast_cb_p(void)
+{
+	int ret;
+
+	ret = task_manager_unset_broadcast_cb(TM_BROADCAST_WIFI_ON, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unset_broadcast_cb", ret, TM_UNREGISTERED_MSG);
+
+	ret = task_manager_unset_broadcast_cb(tm_broadcast_undefined_msg, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unset_broadcast_cb", ret, OK);
```

Comment:
You tried to check undefined_msg. Is it correct that return is OK?

Reply:
Yes, becasue, at utc_task_manager_set_broadcast_cb_p(), the broadcast cb is set with tm_broadcast_undefined_msg.